### PR TITLE
Cleanup duplicate code in the lambda authorizers

### PR DESCRIPTION
**Pull request by jmkulwik**
Quick cleanup to move the duplicate code shared between the two authorizer lambdas into its own module.

Follow-up to a few comments on this pr: https://github.com/code-dot-org/javabuilder/pull/224/files 

Task: [JAVA-471](https://codedotorg.atlassian.net/browse/JAVA-471)

### CODE REVIEW

File: javabuilder-authorizer/http_authorizer_function.rb [hunk 1]
```diff
@@ -1,109 +1,18 @@
...
 require 'jwt'
+require_relative 'jwt_helper'
+include JwtHelper
 
-# The lambda handler takes an event with the query string parameter
-# 'Authorization=token',  where the token is a JWT token generated
-# by dashboard. The handler verifies the token with the appropriate
-# public key (dependant on the environment the request came from),
-# and checks the token has not expired and its issue time is not in the future.
+# The lambda handler takes an event with the query string parameter 'Authorization=token',
```
Comment (sanchitmalhotra126) [inline javabuilder-authorizer/http_authorizer_function.rb:6]: nit: could you add to this documentation (and to the equivalent in the websocket authorizer) to differentiate the two authorizers?

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice!

File: javabuilder-authorizer/jwt_helper.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  # We only care about stage--that tells us the environment (development, staging, etc.)
+  # def get_public_key(route_arn)
+  def get_public_key(origin)
+    # Temporarily choose the key based on the client origin rather than the
+    # route_arn until we have environment-specific Javabuilders set up.
+    # tmp = route_arn.split(':')
+    # api_gateway_arn = tmp[5].split('/')
+    # stage_name = api_gateway_arn[1]
+
+    # TODO: Update the following strings to use the exact origin that requests come from.
```
Comment (molly-moen) [inline javabuilder-authorizer/jwt_helper.rb:70]: do we have a work item for this task?

Comment (jmkulwik) [reply javabuilder-authorizer/jwt_helper.rb:70]: Yeah, it's the second half of this task: https://codedotorg.atlassian.net/browse/JAVA-471

File: javabuilder-authorizer/http_authorizer_function.rb (hunk 1 above)
Comment (jmkulwik) [reply javabuilder-authorizer/http_authorizer_function.rb:6]: Done. Thanks!

**Pull request merged**